src/jit-format/jit-format.cs

```diff
@@ -706,62 +708,100 @@ public static bool RunClangTidy(List<string> filenames, string compileCommands,
+            string formatFix = fix ? "-i" : "";
+            string outputReplacementXml = fix ? "" : "-output-replacements-xml";
             bool formatOk = true;
 
-            foreach (string filename in filenames)
-            {
-                Process process = new Process();
-
-                if (verbose)
+            Parallel.ForEach(filenames, (filename) =>
                 {
-                    Console.WriteLine("Running: clang-format {0} -style=file {1}", formatFix, filename);
-                }
-
-                process.StartInfo.FileName = "clang-format";
-                process.StartInfo.Arguments = "-style=file " + formatFix + filename;
-                process.StartInfo.UseShellExecute = false;
-                process.StartInfo.RedirectStandardOutput = true;
-                process.StartInfo.CreateNoWindow = true;
-                process.Start();
-
-                List<string> fixedContents = new List<string>();
+                    Process process = new Process();
 
-                if (!fix)
-                {
-                    while (!process.StandardOutput.EndOfStream)
+                    if (verbose)
                     {
-                        string line = process.StandardOutput.ReadLine();
-                        fixedContents.Add(line);
+                        Console.WriteLine("Running: clang-format {0} -style=file {1}", formatFix, filename);
                     }
 
-                    int i = 0;
-                    // Diff fixedContents and contents of actual file
-                    foreach (string line in File.ReadLines(filename))
+                    // Run clang-format
+                    List<string> commandArgs = new List<string> { formatFix, "-style=file", outputReplacementXml, filename };
+                    CommandResult result = TryCommand("clang-format", commandArgs, true);
+
+                    if (result.StdOut.Contains("<replacement ") && !fix)
                     {
-                        if (i >= fixedContents.Count || line != fixedContents[i])
+                        if (verbose)
                         {
-                            if (verbose)
+                            // Read in the file
+                            string fileContents = File.ReadAllText(filename);
+                            string[] fileContentsList = fileContents.Split(new[] { Environment.NewLine }, StringSplitOptions.None);
+                            int offsetChange = 0;
+
+                            // Read in the xml generated by clang-format
+                            var doc = new XmlDocument();
+                            doc.LoadXml(result.StdOut);
+
+                            XmlNodeList replacements = doc.DocumentElement.ChildNodes;
+
+                            Console.WriteLine("");
+                            Console.WriteLine("clang-format: there are formatting errors in {0}", filename);
+
+                            foreach (XmlNode replacement in replacements)
```

Comment:
I think probably not. The nice thing about this change is that we can actually show the user all of the formatting errors that they have in each file, not just the first formatting error.